asteria/src/global_context.hpp

```diff
@@ -0,0 +1,27 @@
+// This file is part of Asteria.
+// Copyleft 2018, LH_Mouse. All wrongs reserved.
+
+#ifndef ASTERIA_GLOBAL_CONTEXT_HPP_
+#define ASTERIA_GLOBAL_CONTEXT_HPP_
+
+#include "fwd.hpp"
+#include "abstract_context.hpp"
+
+namespace Asteria {
+
+class Global_context : public Abstract_context
```

Comment:
Just a question: have you designed global context with dynamic-scoped variables or thread_local in mind?

Reply:
The global context is used to share data between **different files**. There is no syntactical construction to declare a new named reference in it. As a result, references in the global context can only be created via binding functions. Its main purpose was the `import` directive which was only barely in my mind and now (I think) is superfluous. The `require` operator (like in _Node.js_ and _Lua_) would be preferred, which, nevertheless, still requires a global context in order to detect circular dependencies.